src/mixins/java/org/spongepowered/common/mixin/core/service/permission/SubjectMixin.java

```diff
@@ -70,15 +70,14 @@ private void subjectConstructor(final CallbackInfo ci) {
     @Override
     public Optional<SubjectReference> bridge$resolveReferenceOptional() {
         if (this.impl$subjectReference == null) {
-            final Optional<PermissionService> serv = SpongeCommon.getGame().getServiceManager().provide(PermissionService.class);
-            serv.ifPresent(permissionService -> new SubjectSettingCallback(this).test(permissionService));
+            new SubjectSettingCallback(this).test(SpongeCommon.getGame().getServiceProvider().permissionService());
```

Comment:
Now that callbacks are gone can this be simplified to a bipredicate (as a lambda I mean) ?

Reply:
More likely to be a static method, the callback isn't really a callback anymore.